src/Core/AdminConsole/Services/Implementations/OrganizationService.cs

```diff
@@ -1913,17 +1930,21 @@ await _organizationUserRepository.DeleteManyAsync(removeUsersSet
 
             var newGroups = groups
                 .Where(g => !existingExternalGroupsDict.ContainsKey(g.Group.ExternalId))
-                .Select(g => g.Group);
+                .Select(g => g.Group).ToList();
 
+            var savedGroups = new List<Group>();
```

Comment:
It took me a moment to figure out why you're compiling this separate list. I assume it's because `groupRepository.CreateAsync` could return a new object?

Reply:
It's because this
```c#
            var newGroups = groups
                .Where(g => !existingExternalGroupsDict.ContainsKey(g.Group.ExternalId))
                .Select(g => g.Group).ToList();

            foreach (var group in newGroups)
            {
                group.CreationDate = group.RevisionDate = DateTime.UtcNow;
                newGroups[newGroups.IndexOf(group)] = await _groupRepository.CreateAsync(group);
                await UpdateUsersAsync(group, groupsDict[group.ExternalId].ExternalUserIds,
                    existingExternalUsersIdDict);
            }
```

Does this

```
    System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
         at System.Collections.Generic.List`1.Enumerator.MoveNext()
         at Bit.Core.Services.OrganizationService.ImportAsync(Guid organizationId, Nullable`1 importingUserId, IEnumerable`1 groups, IEnumerable`1 newUsers, IEnumerable`1 removeUserExternalIds, Boolean overwriteExisting, Nullable`1 eventSystemUser) in /Users/me/bitwarden/server/src/Core/AdminConsole/Services/Implementations/OrganizationService.cs:line 1936

```

Reply:
Thanks for the explanation!